idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/column.py

```diff
@@ -677,6 +688,10 @@ def rule_mass_transfer_coeff_vap(blk, t, x, j):
                       blk.vapor_phase.properties[t, x].dens_mass) /
                      (blk.a_ref *
                       blk.vapor_phase.properties[t, x].visc_d))**(3/4))
+            elif x == self.vapor_phase.length_domain.first():
```

Comment:
This check should (generally) always go first - we don't need to write terms at the inlet, so we should check first. Otherwise, the above condition will be met for at least some species and thus terms will be written at the boundary for them.